### PR TITLE
added charm-release github action

### PR DESCRIPTION
**Pull request by gabrielcocenza**

### CODE REVIEW

File: .github/workflows/charm-release.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Release Charm
+
+on:
+  workflow_call:
+    inputs:
```
Comment (rgildein) [inline .github/workflows/charm-release.yaml:5]: I think we need to add also `channel` and `upload-image` arguments here.

File: .github/workflows/charm-release.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      artifact:
+        description: "Name of artifact to download before building. Must contain the file artifact.tar.gz."
+        default: ''
+        required: false
+        type: string
+    secrets:
+      CHARMHUB_TOKEN:
```
Comment (rgildein) [inline .github/workflows/charm-release.yaml:12]: Is this really needed, since `release-charm` job required it. What is difference here?

**Review by rgildein — CHANGES_REQUESTED** [review]
Overall LGTM, I think we need to only add `channel` and `upload-image` inputs to charm-release.yaml. There is also one question about secretes.

File: .github/workflows/charm-release.yaml (hunk 2 above)
Comment (gabrielcocenza) [reply .github/workflows/charm-release.yaml]: Good catch. I guess you meant that file `_charm-release.yaml` already it is declaring `CHARMHUB_TOKEN` and there is no reason for declaring here again, right?  Observability team used like this an I didn't  contest.

I've removed from here and it still works. I guess setting `secrets: inherit` does a good job.

File: .github/workflows/pull-request.yaml [hunk 3]
```diff
@@ -54,11 +54,10 @@ jobs:
   lint-unit:
     name: Lint checkers and unit tests
     uses: ./.github/workflows/_lint-unit.yaml
-    # Note(rgildein): Lint checkers and unit tests should be triggered on any push, any
-    # pull requested, approval pull request review was submitted or review comment with
+    # Note(rgildein): Lint checkers and unit tests should be triggered on any push (charm-release),
```
Comment (rgildein) [inline .github/workflows/pull-request.yaml:57]: I think that we can remove any mention about push and triggered this workflow only in pull_request and pull_request_review. Can I ask you to remove this [line](https://git.launchpad.net/bootstack-charms-spec/tree/templates/charm/.github/workflows/pr.yaml.j2#n5) from our templates.

**Review by rgildein — COMMENTED** [review]
LGTM, but I would prefer to drop any mention about push in pull-request.yaml

**Review by rgildein — APPROVED** [review]
LGTM, thanks for this

**Review by Pjack — APPROVED** [review]
LGTM

File: .github/workflows/charm-release.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        description: |
+          Command to run functional test in strategies. It defaults to
+          `make functional`, but is defined as a list, so `['make functional']`.
+          This allows you to run multiple tests with different parameters.
+          Examples:
+          - ['FUNC_ARGS="--series jammy" make functional',
+             'FUNC_ARGS="--series focal" make functional']
+          - ['tox -e func -- -sv --series jammy', 'tox -e func -- -sv --series focal']
+        type: string
+        default: "['make functional']"
```
Comment (rgildein) [inline .github/workflows/charm-release.yaml:54]: Hi @gabrielcocenza, sorry I did not saw this. Can we remove default value here so we will use same approach like in pr.yaml. Thanks

Comment (gabrielcocenza) [reply .github/workflows/charm-release.yaml:54]: See #18

Comment (rgildein) [reply .github/workflows/charm-release.yaml:54]: Thank you :+1:

**Pull request merged**